bundles/org.smarthomej.transform.basicprofiles/src/main/java/org/smarthomej/transform/basicprofiles/internal/profiles/RoundStateProfile.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2021 Contributors to the SmartHome/J project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.smarthomej.transform.basicprofiles.internal.profiles;
+
+import static org.smarthomej.transform.basicprofiles.internal.factory.BasicProfilesFactory.ROUND_UID;
+
+import java.math.RoundingMode;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.thing.profiles.ProfileCallback;
+import org.openhab.core.thing.profiles.ProfileContext;
+import org.openhab.core.thing.profiles.ProfileTypeUID;
+import org.openhab.core.thing.profiles.StateProfile;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.Type;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.smarthomej.transform.basicprofiles.internal.config.RoundStateProfileConfig;
+
+/**
+ * Applies rounding with the specified scale and the rounding mode to a {@link QuantityType} or {@link DecimalType}
+ * state. Default rounding mode is {@link RoundingMode#HALF_UP}.
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+@NonNullByDefault
+public class RoundStateProfile implements StateProfile {
+
+    private final Logger logger = LoggerFactory.getLogger(RoundStateProfile.class);
+
+    private final ProfileCallback callback;
+
+    final int scale;
+    final RoundingMode roundingMode;
+
+    public RoundStateProfile(ProfileCallback callback, ProfileContext context) {
+        this.callback = callback;
+        RoundStateProfileConfig config = context.getConfiguration().as(RoundStateProfileConfig.class);
+        logger.debug("Configuring profile with parameters: [{scale='{}', mode='{}']", config.scale, config.mode);
+
+        int localScale = 0;
+        // if (config.scale instanceof String) {
+        // try {
+        // scale = Integer.valueOf((String) config.scale);
+        // } catch (NumberFormatException e) {
+        // logger.error("Cannot convert value '{}' of parameter 'scale' into a valid integer.", config.scale);
+        // }
+        // } else
+        Integer configScale = config.scale;
+        if (configScale instanceof Number) {
+            localScale = ((Number) configScale).intValue();
+        } else {
+            logger.error("Parameter 'scale' is not of type String or Number.");
+        }
+
+        if (localScale < 0) {
```

Comment:
Why is this restricted to non-negative numbers? I always thought that a negative scale is allowed and e.g. a scale of -2 round 1234 to 1200. This could also be useful.

Reply:
Can you comment here?

Reply:
Sry. I missed this comment. Might be useful too. Will change it.

Reply:
Done.